src/scenes/wizards_waltz.inc.c

```diff
@@ -1,37 +1,316 @@
+void waltz_main_unload(void) { /* Stub Function */ }
+
+
+// !TODO - Currently impossible without an ASM hack. See: https://decomp.me/scratch/Kk2Ec
+// [func_08044e78] CUE Behaviour - Spawn
 #include "asm/scenes/wizards_waltz/asm_08044e78.s"
 
-#include "asm/scenes/wizards_waltz/asm_08044f94.s"
 
-#include "asm/scenes/wizards_waltz/asm_08044fc0.s"
+// [func_08044f94] CUE Behaviour - Unknown
+u32 waltz_cue_unknown(u32 arg0, u32 arg1, u32 arg2) {
+    if (arg2 > (GameInfo.cycleInterval + func_0800c3a4(0x30))) {
+        return 1; }
+    else {
+        return 0; }
+}
+
+
+// [func_08044fc0] CUE Behaviour - Despawn
+void waltz_cue_despawn(u32 arg0, u32 *arg1) {
+    func_0800fc70(arg1[0]);
+}
+
+
+// [func_08044fcc] CUE Event - Hit
+void waltz_cue_hit(u32 arg0, struct struct_080179f4_sub *arg1) {
+    u32 isTutorial;
+
+    // Play animation: "sprout_grow"
+    func_08010064(arg1->unk0, D_08932d2c, 0, 1, 0x7f, 0);
+
+    // Cycle through frames of "girl_happy" if isTutorial flag is not set.
+    isTutorial = GameInfo.isTutorial;
+    if (!isTutorial) {
+        GameInfo.girlState = 1;
+
+        // Increment frame for "girl_happy" if not at the maximum of 5.
+        if (GameInfo.flowerCount <= 5) {
+            GameInfo.flowerCount += 1;
+        }
+
+        // Play animation: "girl_happy"
+        func_08010064(GameInfo.girlEntity, D_08932e64, GameInfo.flowerCount, 0, isTutorial, isTutorial);
+    }
+}
+
+
+// [func_0804503c] CUE Event - Barely
+void waltz_cue_barely(u32 arg0, struct struct_080179f4_sub *arg1) {
+    u32 temp;
+    u32 isTutorial;
+
+    // Set unknown parameter.
+    temp = ((u32) ~func_08018054()) >> 0x1f;
+    if (arg1->unk6 > 0x200) {
+        temp ^= 1;
+    }
+
+    // Play animation: "sprout_eaten"
+    func_08010064(arg1->unk0, D_08932cbc, 0, 1, 0x7f, 0);
+    // Unknown function.
+    func_0800ff44(arg1->unk0, temp);
+
+    // Cycle through frames of "girl_sad" if isTutorial flag is not set.
+    isTutorial = GameInfo.isTutorial;
+    if (!isTutorial) {
+        GameInfo.girlState = 2;
+
+        // Decrement frame for "girl_sad" if not at the minimum of 0.
+        if (GameInfo.flowerCount) {
+            GameInfo.flowerCount -= 1;
+        }
+
+        // Play animation: "girl_sad"
+        func_08010064(GameInfo.girlEntity, D_08932ea4, GameInfo.flowerCount, 0, isTutorial, isTutorial);
+    }
+
+    // Unknown function - likely related to score.
+    func_0800bc40();
+}
+
+
+// [func_080450d0] CUE Event - Miss
+void waltz_cue_miss(u32 arg0, struct struct_080179f4_sub *arg1) {
+    // Unknown function - likely related to score.
+    func_0800bc40();
+}
+
 
-#include "asm/scenes/wizards_waltz/asm_08044fcc.s"
+// [func_080450dc] MAIN - Input Event
+void wizard_main_unknown(void) { /* Stub Function */ }
 
-#include "asm/scenes/wizards_waltz/asm_0804503c.s"
 
-#include "asm/scenes/wizards_waltz/asm_080450d0.s"
+// [func_080450e0] OTHER - Unknown Func_00
+void func_080450e0(void) { /* Stub Function */ }

```

Comment:
For stub functions like this, you can name them "filename_stub_x" (where x increments for each stub) so this could be "wizards_waltz_stub_1".